docs/src/developing/clients/jsonrpc-api.md

```diff
@@ -2444,6 +2445,96 @@ The first leader returned is the leader for slot #100:
 }
 ```
 
+### getSnapshotInfo
```

Comment:
Will the caller know whether they are requesting information about a full or incremental snapshot?  If so I suspect that adding two RPC methods instead of one will be simpler from an implementation and interface perspective.

Reply:
Yeah, that's kinda the crux of this PR.

During bootstrap, the node (caller) gets a `(Slot, Hash)` for a snapshot to download, but it doesn't know if that corresponds to a full snapshot or an incremental snapshot. The caller needs to know the snapshot type in order to (1) call `download_snapshot_archive()` with the right params, and (2), if the type is an incremental snapshot, the caller may also need to download the associated full snapshot too.

So this RPC fn is returning info to answer both questions (1) and (2) for the caller. 

Maybe there's another way? This was just one way that I thought of doing it, but happy to change to something better!

Reply:
> During bootstrap, the node (caller) gets a (Slot, Hash) for a snapshot to download, but it doesn't know if that corresponds to a full snapshot or an incremental snapshot

I see. It gets this `(Slot, Hash)` from gossip right?

Reply:
`CrdsData:: SnapshotHashes` specifically?  The reason I ask is because I think we probably need a new CRDS data type to share incremental snapshot hashes over gossip for backwards compatibility, either by complimenting or deprecating `CrdsData:: SnapshotHashes`.  Once we do that the client will then know from a `(Slot, Hash)` whether the snapshot is full or incremental and won't need to ask RPC at all.

Reply:
> > During bootstrap, the node (caller) gets a (Slot, Hash) for a snapshot to download, but it doesn't know if that corresponds to a full snapshot or an incremental snapshot
> 
> I see. It gets this `(Slot, Hash)` from gossip right?

Yep!